scripts/ci/release-upgrade-canary.py

```
@@ -0,0 +1,741 @@
+                    candidate=False,
+                ) as previous:
+                    for prompt in PROMPTS:
+                        thread_id = _create_thread(previous.base_url)
+                        _send_message(previous.base_url, thread_id, prompt)
+                        _wait_for_complete_timeline(previous.base_url, thread_id)
+                    baseline = capture_snapshot(previous.base_url)
+                if len(baseline.thread_ids) != len(PROMPTS) or baseline.message_count != 4:
+                    raise CanaryFailure(
+                        "previous artifact did not create the required two-thread/four-message baseline: "
+                        f"threads={len(baseline.thread_ids)} messages={baseline.message_count}"
+                    )
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)